src/node_crypto.cc

```diff
@@ -745,6 +747,73 @@ void SecureContext::SetKey(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void SecureContext::SetEngineKey(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  SecureContext* sc;
+  ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
+
+  unsigned int len = args.Length();
+  if (len != 2) {
+    return THROW_ERR_MISSING_ARGS(env,
+                 "Private key name and engine name are required");
```

Comment:
If this function is not supposed to be called directly, use `CHECK_EQ(len, 2)` instead. I know that the TLS code base is not a great example, but this kind of input validation should not happen in C++.